src/main/java/com/checkmarx/flow/service/MergeResultEvaluatorImpl.java

```diff
@@ -30,13 +27,19 @@ public MergeResultEvaluatorImpl(FlowProperties flowProperties) {
     @Override
     public boolean isMergeAllowed(ScanResults scanResults, RepoProperties repoProperties) {
         if (!repoProperties.isErrorMerge()) {
+            log.info("Merge is allowed, because error-merge is set to false.");
             return true;
         }
 
+        log.debug("Checking if pull request merge is allowed.");
         Map<FindingSeverity, Integer> thresholds = getEffectiveThresholds();
-        writeToLog(thresholds);
+        writeMapToLog(thresholds, "Using thresholds");
 
-        return !isAnyThresholdExceeded(scanResults, thresholds);
+        boolean isAllowed = !isAnyThresholdExceeded(scanResults, thresholds);
+        log.info(isAllowed ? "Merge is allowed, because no thresholds were exceeded." :
+                "Merge is not allowed, because some of the thresholds were exceeded.");
+
+        return isAllowed;
```

Comment:
Something Aviv asked me - in case we apply the threshold - can we print to the log why we applied it against the actual findings were found?